server/server.ts

```diff
@@ -0,0 +1,45 @@
+require('dotenv').config();
+import { ApolloServer } from 'apollo-server';
+import { connectDB } from "./mongoConnection";
+
+import { config } from './config';
+
+import { Request, RequestDocument } from './models/requestModel';
+import RequestDataSource from './datasources/requestsDataSource'
+import { RequestsCache } from "./cache";
+
+// TODO: need to make script to build(compile) prod server and to run prod server
+
+
+// //-----------------------------------------------------------------------------
+// // APOLLO SETUP
+// //-----------------------------------------------------------------------------
+
+import { typeDefs } from "./schema";
+import { resolvers } from "./resolvers";
+
+//-----------------------------------------------------------------------------
+// MONGODB CONNECTION AND DATA SOURCES FOR APOLLO
+//-----------------------------------------------------------------------------
+
+// connect to MongoDB and setup data sources
+connectDB(() => {
+    if (config.caching) {
+        RequestsCache.init();
```

Comment:
Do we just update the cache whenever we mutate a request or add a new request?

Reply:
[https://github.com/uwblueprint/pregnancy-centre/blob/aeb9efd14c8950d967f6752a07e129f86a55b32a/server/cache.ts#L30](url)

Here, we invoke a change stream, which watches whenever mongoose performs any change (think writing to) to the database. 

In the case that this isn't enough for future uses, it would be easy to implement an update() method for Cache that can be called when ever we do any change to the database ourselves.

Reply:
👍👍 answers my question perfectly, thanks!